File: fm-dddrive/src/main/java/io/dddrive/core/oe/model/base/ObjTenantBase.java
package io.dddrive.core.oe.model.base;

import java.time.OffsetDateTime;

import io.dddrive.core.obj.model.ObjRepository;
import io.dddrive.core.obj.model.base.ObjBase;
import io.dddrive.core.oe.model.ObjTenant;
import io.dddrive.core.property.model.BaseProperty;

public abstract class ObjTenantBase extends ObjBase implements ObjTenant {

	//@formatter:off
	protected final BaseProperty<String> key = this.addBaseProperty("key", String.class);
	protected final BaseProperty<String> name = this.addBaseProperty("name", String.class);
	protected final BaseProperty<String> description = this.addBaseProperty("description", String.class);
	//@formatter:on

	public ObjTenantBase(ObjRepository<ObjTenant> repository) {
		super(repository);
	}

//	@Override
//	public void doCalcSearch() {
//		super.doCalcSearch();
//		this.addSearchText(this.getName());
//		this.addSearchText(this.getDescription());
//	}

	@Override
	public void doAfterCreate(Object userId, OffsetDateTime timestamp) {
		super.doAfterCreate(userId, timestamp);
		this.tenant.setId(this.getId());
	}

	@Override
	protected void doCalcAll() {
		super.doCalcAll();
		this.calcCaption();
	}

	protected void calcCaption() {
		this.setCaption(this.getName());
	}

}

File: fm-dddrive/src/main/java/io/dddrive/core/oe/model/base/ObjTenantBase.kt
package io.dddrive.core.oe.model.base

import io.dddrive.core.obj.model.ObjRepository
import io.dddrive.core.obj.model.base.ObjBase
import io.dddrive.core.oe.model.ObjTenant
import io.dddrive.path.setValueByPath
import java.time.OffsetDateTime

abstract class ObjTenantBase(
	override val repository: ObjRepository<ObjTenant>,
	isNew: Boolean,
) : ObjBase(repository, isNew),
	ObjTenant {

	override fun doInit() {
		super.doInit()
		addBaseProperty<String>("key", String::class.java)
		addBaseProperty<String>("name", String::class.java)
		addBaseProperty<String>("description", String::class.java)
	}

	override fun doAfterCreate(
		userId: Any,
		timestamp: OffsetDateTime,
	) {
		super.doAfterCreate(userId, timestamp)
		setValueByPath("tenantId", id)
	}

	override fun doCalcAll() {
		super.doCalcAll()
		calcCaption()
	}

	protected fun calcCaption() {
		setCaption(name)
	}

	// 	@Override
	// 	public void doCalcSearch() {
	// 		super.doCalcSearch();
	// 		addSearchText(getName());
	// 		addSearchText(getDescription());
	// 	}

}
